tests/models.py

```diff
@@ -41,8 +41,8 @@ def formfield(self, **kwargs):
 
 class User(models.Model):
     username = models.CharField(_('username'), max_length=255)
-    first_name = SubCharField(max_length=100)
-    last_name = SubSubCharField(max_length=100)
+    first_name = SubCharField(max_length=100, null=True, blank=True)
+    last_name = SubSubCharField(max_length=100, null=True, blank=True)
```

Comment:
I believe, by default an empty string is null. Also, you are not using forms, so `blank=True` is not needed. hence, you don't need to touch these two lines.

Reply:
> I believe, by default an empty string is null

I am not sure what you mean. `null` and empty string are different values and are treated differently both by Django ORM and the underlying database.

`null=True` is necessary because within tests I create some User instances with first_name=null to check handling of nulls. The default value for the `null` constructor parameter is False.

As for `blank=True`, maybe User instances are not tied to forms at the moment, but they can be in the future. I think it's a good practice to set `blank=True` once one defines `null=True`.

Reply:
This is from the Django official doc: https://docs.djangoproject.com/en/3.0/ref/models/fields/#django.db.models.Field.null

Reply:
> This is from the Django official doc: https://docs.djangoproject.com/en/3.0/ref/models/fields/#django.db.models.Field.null

Still, having `null=False` will prevent us from testing `User(first_name=null)`

> As for the blank=True, I believe it should be added on demand. Otherwise, me reading the tests would scratch my head and think why is it necessary to have it here?

Well, I would rather scratch my head if I saw a field with null=True but blank=False. It clearly won't work with forms at all. I guess we need another opinion here, ideally from django-filter devs.

Reply:
I have some experience with Django and would like to share my opinion on this matter. I think having blank=True, null=True on a django CharField is definitely not a best practice and even strongly disencouraged. It means you have two empty values in the database, NULL and '' (empty char). Therefore my recommendation is to rewrite the test by using a datetimeField (for example) with null=True, blank=True. In that case it makes logical sense and the correctness of the test is also faily easily determined.

Having said that I would also love to see this merged! Big applause for @earshinov for undertaking this project!

Reply:
Hi @gabn88! Sorry that I didn't have time to reply during workdays. Thank you for your input, your suggestion about avoiding nullable/blank CharFields seems very reasonable. I reworked the tests using a DateTime model field instead of CharField and rebased the branch onto the latest `master` along the way. Hopefully these changes will help merge this PR.